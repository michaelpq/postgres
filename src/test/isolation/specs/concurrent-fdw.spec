# Test for interactions with DDL commands manipulating fdw
# objects.
# The following set of commands can interact in concurrency:
# - ALTER FDW OPTIONS
# - ALTER FDW VALIDATOR
# - ALTER FDW HANDLER
# - ALTER FDW OWNER TO

setup
{
  CREATE FOREIGN DATA WRAPPER regress_fdw;
}

teardown
{
  DROP FOREIGN DATA WRAPPER regress_fdw;
}

session "s1"
step "s1_begin"			{ BEGIN; }
step "s1_options"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1'); }
step "s1_validator"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR; }
step "s1_handler"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER; }
step "s1_owner"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER; }
step "s1_commit"	{ COMMIT; }

session "s2"
step "s2_begin"			{ BEGIN; }
step "s2_options"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1'); }
step "s2_validator"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR; }
step "s2_handler"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER; }
step "s2_owner"	{
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER; }
step "s2_commit"	{ COMMIT; }

permutation "s1_begin" "s1_options" "s2_begin" "s2_options"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_options" "s2_begin" "s2_validator"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_options" "s2_begin" "s2_handler"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_options" "s2_begin" "s2_owner"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_validator" "s2_begin" "s2_validator"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_validator" "s2_begin" "s2_handler"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_validator" "s2_begin" "s2_owner"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_handler" "s2_begin" "s2_handler"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_handler" "s2_begin" "s2_owner"
	"s1_commit" "s2_commit"
permutation "s1_begin" "s1_owner" "s2_begin" "s2_owner"
	"s1_commit" "s2_commit"
