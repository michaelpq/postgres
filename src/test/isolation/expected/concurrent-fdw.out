Parsed test spec with 2 sessions

starting permutation: s1_begin s1_options s2_begin s2_options s1_commit s2_commit
step s1_begin: BEGIN;
step s1_options: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1');
step s2_begin: BEGIN;
step s2_options: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1'); <waiting ...>
step s1_commit: COMMIT;
step s2_options: <... completed>
error in steps s1_commit s2_options: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;

starting permutation: s1_begin s1_options s2_begin s2_validator s1_commit s2_commit
step s1_begin: BEGIN;
step s1_options: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1');
step s2_begin: BEGIN;
step s2_validator: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR; <waiting ...>
step s1_commit: COMMIT;
step s2_validator: <... completed>
error in steps s1_commit s2_validator: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;

starting permutation: s1_begin s1_options s2_begin s2_handler s1_commit s2_commit
step s1_begin: BEGIN;
step s1_options: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1');
step s2_begin: BEGIN;
WARNING:  changing the foreign-data wrapper handler can change behavior of existing foreign tables
step s2_handler: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER; <waiting ...>
step s1_commit: COMMIT;
step s2_handler: <... completed>
error in steps s1_commit s2_handler: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;

starting permutation: s1_begin s1_options s2_begin s2_owner s1_commit s2_commit
step s1_begin: BEGIN;
step s1_options: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OPTIONS (a '1');
step s2_begin: BEGIN;
step s2_owner: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER;
step s1_commit: COMMIT;
step s2_commit: COMMIT;

starting permutation: s1_begin s1_validator s2_begin s2_validator s1_commit s2_commit
step s1_begin: BEGIN;
step s1_validator: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR;
step s2_begin: BEGIN;
step s2_validator: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR; <waiting ...>
step s1_commit: COMMIT;
step s2_validator: <... completed>
error in steps s1_commit s2_validator: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;

starting permutation: s1_begin s1_validator s2_begin s2_handler s1_commit s2_commit
step s1_begin: BEGIN;
step s1_validator: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR;
step s2_begin: BEGIN;
WARNING:  changing the foreign-data wrapper handler can change behavior of existing foreign tables
step s2_handler: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER; <waiting ...>
step s1_commit: COMMIT;
step s2_handler: <... completed>
error in steps s1_commit s2_handler: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;

starting permutation: s1_begin s1_validator s2_begin s2_owner s1_commit s2_commit
step s1_begin: BEGIN;
step s1_validator: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO VALIDATOR;
step s2_begin: BEGIN;
step s2_owner: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER;
step s1_commit: COMMIT;
step s2_commit: COMMIT;

starting permutation: s1_begin s1_handler s2_begin s2_handler s1_commit s2_commit
step s1_begin: BEGIN;
WARNING:  changing the foreign-data wrapper handler can change behavior of existing foreign tables
step s1_handler: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER;
step s2_begin: BEGIN;
WARNING:  changing the foreign-data wrapper handler can change behavior of existing foreign tables
step s2_handler: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER; <waiting ...>
step s1_commit: COMMIT;
step s2_handler: <... completed>
error in steps s1_commit s2_handler: ERROR:  tuple concurrently updated
step s2_commit: COMMIT;

starting permutation: s1_begin s1_handler s2_begin s2_owner s1_commit s2_commit
step s1_begin: BEGIN;
WARNING:  changing the foreign-data wrapper handler can change behavior of existing foreign tables
step s1_handler: 
	ALTER FOREIGN DATA WRAPPER regress_fdw NO HANDLER;
step s2_begin: BEGIN;
step s2_owner: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER;
step s1_commit: COMMIT;
step s2_commit: COMMIT;

starting permutation: s1_begin s1_owner s2_begin s2_owner s1_commit s2_commit
step s1_begin: BEGIN;
step s1_owner: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER;
step s2_begin: BEGIN;
step s2_owner: 
	ALTER FOREIGN DATA WRAPPER regress_fdw OWNER TO CURRENT_USER;
step s1_commit: COMMIT;
step s2_commit: COMMIT;
